### PR TITLE
hybrid at core

### PR DESCRIPTION
**Pull request by sasadep**
* bare minimum hybrid infrastructure. 
* currently using backend unit test for testing.

### CODE REVIEW

File: src/ngraph/runtime/hybrid/hybrid_backend.cpp [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+shared_ptr<runtime::Backend> runtime::hybrid::HYBRIDBackend::get_cached_backend(Placement placement)
+{
+    if (m_cached_backends.find(placement) == m_cached_backends.end())
+    {
+        m_cached_backends[placement] = runtime::Backend::create(placement_to_string(placement));
+    }
+    return m_cached_backends.at(placement);
+}
+
+shared_ptr<runtime::Tensor> runtime::hybrid::HYBRIDBackend::create_tensor(const element::Type& type,
```
Comment (rkimballn1) [inline src/ngraph/runtime/hybrid/hybrid_backend.cpp]: This function worries me. What if this is a GPU tensor? You need to create the tensor on the proper device but I'm not sure how right now.

File: src/ngraph/runtime/hybrid/hybrid_backend.cpp [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+            Placement placement = get_colocated_function_placement(sub_function);
+            auto backend = get_cached_backend(placement);
+            backend->compile(sub_function);
+        }
+    }
+    NGRAPH_INFO << "hybrid compile -End ";
+    return true;
+}
+
+bool runtime::hybrid::HYBRIDBackend::call(shared_ptr<Function> function,
```
Comment (rkimballn1) [inline src/ngraph/runtime/hybrid/hybrid_backend.cpp]: This entire function is too inefficient. Too many new objects created. We have to make this work with minimal new objects and copies.

File: src/ngraph/runtime/hybrid/hybrid_backend.cpp (hunk 1 above)
Comment (diyessi) [reply src/ngraph/runtime/hybrid/hybrid_backend.cpp]: What if there is no proper device, i.e. the tensor is used on more than one device?

**Comment by rkimballn1** [comment]
This does not pass tests if you don't build the CPU backend. I normally don't build the CPU backend.

File: src/ngraph/runtime/backend.cpp [hunk 3]
```diff
@@ -107,3 +107,10 @@ void runtime::Backend::validate_call(shared_ptr<const Function> function,
         }
     }
 }
+
+bool runtime::Backend::is_supported(const Node& node) const
+{
```
[the remark this answers is not in the document]
Comment (rkimballn1) [reply src/ngraph/runtime/backend.cpp]: I was only making it behave they way it does today. If we make it false then it operates differently than current backends.

**Comment by sasadep** [comment]
@rkimballn1 hybrid needs at least two backends to work in its current state 
CPU and Interpreter is the first two that I started with.

**Pull request merged**